solr/core/src/java/org/apache/solr/cluster/placement/plugins/AffinityPlacementConfig.java

```diff
@@ -43,14 +46,30 @@
   @JsonProperty
   public long prioritizedFreeDiskGB;
 
+  /**
+   * This property defines an additional constraint that primary collections (keys) should be
+   * located on the same nodes as the secondary collections (values). The plugin will assume
+   * that the secondary collection replicas are already in place and ignore candidate nodes where
+   * they are not already present.
+   */
+  @JsonProperty
+  public Map<String, String> withCollections;
+
   // no-arg public constructor required for deserialization
   public AffinityPlacementConfig() {
     minimalFreeDiskGB = 20L;
```

Comment:
I prefer the no arg constructor here to call the appropriate constructor. That way logic is not replicated (might not be applicable here unless somebody replaces `withCollections = Map.of();` with `withCollections = null;` in a future commit), it looks cleaner and by tracing calls to the most complete constructor all callers are inventoried...